src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs

```diff
@@ -47,94 +49,94 @@ public sealed class SqlServerFhirModel : SqlServerModelInitializer
             EnsureArg.IsNotNull(securityConfiguration?.Value, nameof(securityConfiguration));
             EnsureArg.IsNotNull(logger, nameof(logger));
 
-            _searchParameterDefinitionManager = searchParameterDefinitionManagerResolver();
+            _configuration = configuration;
+            _schemaInitializer = schemaInitializer;
+            _searchParameterDefinitionManager = searchParameterDefinitionManager;
             _securityConfiguration = securityConfiguration.Value;
             _logger = logger;
         }
 
         public short GetResourceTypeId(string resourceTypeName)
         {
-            ThrowIfNotInitialized();
             return _resourceTypeToId[resourceTypeName];
         }
 
         public bool TryGetResourceTypeId(string resourceTypeName, out short id)
         {
-            ThrowIfNotInitialized();
             return _resourceTypeToId.TryGetValue(resourceTypeName, out id);
         }
 
         public string GetResourceTypeName(short resourceTypeId)
         {
-            ThrowIfNotInitialized();
             return _resourceTypeIdToTypeName[resourceTypeId];
         }
 
         public byte GetClaimTypeId(string claimTypeName)
         {
-            ThrowIfNotInitialized();
             return _claimNameToId[claimTypeName];
         }
 
         public short GetSearchParamId(Uri searchParamUri)
         {
-            ThrowIfNotInitialized();
             return _searchParamUriToId[searchParamUri];
         }
 
         public byte GetCompartmentTypeId(string compartmentType)
         {
-            ThrowIfNotInitialized();
             return _compartmentTypeToId[compartmentType];
         }
 
         public bool TryGetSystemId(string system, out int systemId)
         {
-            ThrowIfNotInitialized();
             return _systemToId.TryGetValue(system, out systemId);
         }
 
         public int GetSystemId(string system)
         {
-            ThrowIfNotInitialized();
-
             VLatest.SystemTable systemTable = VLatest.System;
             return GetStringId(_systemToId, system, systemTable, systemTable.SystemId, systemTable.Value);
         }
 
         public int GetQuantityCodeId(string code)
         {
-            ThrowIfNotInitialized();
-
             VLatest.QuantityCodeTable quantityCodeTable = VLatest.QuantityCode;
             return GetStringId(_quantityCodeToId, code, quantityCodeTable, quantityCodeTable.QuantityCodeId, quantityCodeTable.Value);
         }
 
         public bool TryGetQuantityCodeId(string code, out int quantityCodeId)
         {
-            ThrowIfNotInitialized();
             return _quantityCodeToId.TryGetValue(code, out quantityCodeId);
         }
 
-        protected override async Task Initialize()
+        public void Start()
         {
-            var connectionStringBuilder = new SqlConnectionStringBuilder(SqlServerDataStoreConfiguration.ConnectionString);
+            StartAsync().GetAwaiter().GetResult();
```

Comment:
Pulled this from [here](https://docs.microsoft.com/en-us/archive/msdn-magazine/2015/july/async-programming-brownfield-async-development).